web-api/src/calendarApp/CalendarAppAuthenticator.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable functional/prefer-readonly-type */
+import * as jwt from "jsonwebtoken";
+
+import axios, { AxiosInstance, AxiosTransformer } from "axios";
+
+import humps from "humps";
+import qs from "qs";
+
+const DEFAULT_ACCESS_TOKEN_LIFETIME = 600;
+
+export type CalendarAppAuthenticatorOptions = {
+  readonly applicationId: string;
+  readonly privateKey: string;
+  readonly accessTokenLifetimeInSec?: number;
+  /** you can overwrite for testing purposes */
+  readonly baseURL?: string;
+  readonly timeout?: number;
+};
+
+export type AccessToken = {
+  readonly accessToken: string;
+  readonly expireAt: number;
+  readonly tokenType: "Bearer";
+};
+
+export class CalendarAppAuthenticator {
+  readonly api: AxiosInstance;
+  private readonly options: CalendarAppAuthenticatorOptions;
+
+  constructor(options: CalendarAppAuthenticatorOptions) {
+    this.options = options;
+    this.api = axios.create({
+      baseURL: options.baseURL || "https://timetreeapis.com",
+      timeout: options.timeout,
+      paramsSerializer: (params) =>
+        qs.stringify(humps.decamelizeKeys(params), { skipNulls: true }),
+      transformResponse: [
+        ...[axios.defaults.transformResponse].flat(),
+        (data) => humps.camelizeKeys(data),
+      ] as AxiosTransformer[],
+      transformRequest: [
+        (data) => humps.decamelizeKeys(data),
+        ...[axios.defaults.transformRequest].flat(),
+      ] as AxiosTransformer[],
+    });
+  }
+
+  public async getAccessToken(installationId: string) {
+    const response = await this.api.post<AccessToken>(
+      `/installations/${installationId}/access_tokens`,
+      null,
+      {
+        headers: {
+          Authorization: `Bearer ${this.generateToken()}`,
+        },
+      }
+    );
+    return response.data;
+  }
+
+  private generateToken() {
+    const now = Math.floor(new Date().getTime() / 1000);
+    const payload = {
+      iat: now,
+      exp:
```

Comment:
The exp claim identifies the expiration time.
Number of **seconds** from 1970-01-01T00:00:00Z UTC until the specified UTC date/time, ignoring leap seconds.
https://tools.ietf.org/html/rfc7519#section-4.1.4